src/views/User/UserByName.tsx

```diff
@@ -20,64 +20,44 @@ import {get} from 'requests';
 import {errorAlerter} from "misc";
 import * as player_cache from "player_cache";
 import {User} from './User';
+import { RouteComponentProps } from "react-router";
 
 
-interface UserByNameProperties {
-    match: {
-        params: any;
-    };
-}
-
-interface UserByNameState { user_id?: number }
-
-export class UserByName extends React.PureComponent<UserByNameProperties, UserByNameState> {
-    constructor(props) {
-        super(props);
-
-        const user = player_cache.lookup_by_username(props.match.params.username);
-
-        this.state = {
-            user_id: user ? user.id : null
-        };
-    }
-
-    componentDidMount() {
-        if (!this.state.user_id) {
-            this.doFetch(this.props.match.params.username);
-        }
-    }
+type UserByNameProperties = RouteComponentProps<{username: string}>;
 
-    UNSAFE_componentWillReceiveProps(next_props) {
-        if (next_props.match.params.username !== this.props.match.params.username) {
-            const user = player_cache.lookup_by_username(next_props.match.params.username);
+export function UserByName(props: UserByNameProperties) {
+    const username = props.match.params.username;
+    const [user_id, set_user_id] = React.useState<number>(null);
 
-            this.setState({user_id: user ? user.id : null});
-
-            if (!user || !user.id) {
-                this.doFetch(next_props.match.params.username);
-            }
-        }
-    }
-
-    doFetch(username: string) {
-        get("players", {username: username})
-        .then((res) => {
+    const doFetch = async(username: string) => {
+        try {
+            const res = await get("players", { username: username });
             if (res.results.length) {
-                this.setState({
-                    user_id: res.results[0].id
-                });
+                set_user_id(res.results[0].id);
             } else {
-                this.setState({user_id: -1});
+                set_user_id(-1);
             }
-        })
-        .catch(errorAlerter);
-    }
+        } catch (args) {
+            return errorAlerter(args);
+        }
+    };
 
-    render() {
-        if (this.state.user_id) {
-            return <User match={{params: {user_id: this.state.user_id}}}/>;
+    React.useEffect(() => {
+        const user_id = player_cache.lookup_by_username(username)?.id;
+        if (user_id != null) {
+            set_user_id(user_id);
+            return;
         }
-        return null;
+
+        void doFetch(username);
```

Comment:
I'd rather not be in the practice of doing this for silencing those unhandled promise errors, while it may quiet the compiler warnings, uncaught errors will end up at our global `onunhandledrejection` handler which will end up sending it to our remote error capturing service, and we only want bad stuff going there, not stuff that can be handled. I acknowledge that the likelyhood of the above code failing would be minimal (happening only if `errorAlerter` threw an exception after the fetch threw an exception), it's more of a "lets not set that precedent" kind of thing so we're not in that habit in the future where it matters more. Best to be explicit in error handling I feel.

That being said, I'd be fine if you want to have some sort of function you call that takes a promise and adds an ignore catch on it or something to "properly" handle it, like `ignorePromise(doFetch(username))` or whatever.

Reply:
For sure.  I'm not even sure why I made that function `async` if I was just going to ignore the promise 😅.  I think I had experimented with using `player_cache.fetch_by_username` or something.  Anyway, I've updated `doFetch` to be syncronous again, and we handle the rejection in there.